app/src/main/java/org/simple/clinic/di/HttpInterceptorsModule.kt

```diff
@@ -0,0 +1,29 @@
+package org.simple.clinic.di
+
+import dagger.Module
+import dagger.Provides
+import okhttp3.Interceptor
+import okhttp3.logging.HttpLoggingInterceptor
+import org.simple.clinic.BuildConfig
+import org.simple.clinic.analytics.NetworkAnalyticsInterceptor
+import org.simple.clinic.user.LoggedInUserHttpInterceptor
+
+@Module
+class HttpInterceptorsModule {
+
+  @Provides
+  fun providerInterceptors(
+      loggedInInterceptor: LoggedInUserHttpInterceptor,
+      appInfoHttpInterceptor: AppInfoHttpInterceptor,
+      networkAnalyticsInterceptor: NetworkAnalyticsInterceptor
+  ): List<Interceptor> {
```

Comment:
Should we pass in a list of interceptors from outside. Thereby, we can leave this module open for extension but closed for modification?

Reply:
The problem with that is that these interceptors themselves are constructor injected and are dependent on other things in the graph. So if we need to provide alternate interceptors in different sourcesets (as is the case in later commits), doing this becomes tricky.

Reply:
In a later commit, I removed the debug specific branch in favour of an alternate module for the sourceset. Doing this in this manner makes the network module itself (where the interceptors are being used) open for extension.